src/main/java/org/folio/service/mapping/convertor/SrsRecordConvertorService.java

```diff
@@ -0,0 +1,139 @@
+import org.marc4j.*;
+import org.marc4j.marc.Record;
+import org.marc4j.marc.VariableField;
+import org.marc4j.marc.impl.SortedMarcFactoryImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SrsRecordConvertorService extends RecordConvertor {
+  @Autowired
+  private MappingService mappingService;
+
+  @Autowired
+  private RecordLoaderService recordLoaderService;
+  private SortedMarcFactoryImpl sortedMarcFactory = new SortedMarcFactoryImpl();
+  private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup()
+    .lookupClass());
+
+  public List<String> transformSrsRecords(MappingProfile mappingProfile, List<JsonObject> srsRecords, String jobExecutionId,
+      OkapiConnectionParams connectionParams) {
+    if (isTransformationRequired(mappingProfile)) {
+      return transformSrsRecord(mappingProfile, srsRecords, jobExecutionId, connectionParams);
+    } else {
+      return getRecordContent(srsRecords);
+    }
+  }
+
+  private boolean isTransformationRequired(MappingProfile mappingProfile) {
+    List<Transformations> transformations = mappingProfile.getTransformations();
+    List<RecordType> recordTypes = mappingProfile.getRecordTypes();
+    return isNotEmpty(transformations) && (recordTypes.contains(RecordType.HOLDINGS) || recordTypes.contains(RecordType.ITEM));
+
+  }
+
+  public List<String> transformSrsRecord(MappingProfile mappingProfile, List<JsonObject> srsRecords, String jobExecutionId,
+      OkapiConnectionParams connectionParams) {
+    List<String> marcRecords = new ArrayList<>();
+    for (JsonObject srsRecord : srsRecords) {
+      // generate record fields by mapping profile
+      List<VariableField> mappedFields = getMappedFields(mappingProfile, jobExecutionId, connectionParams, srsRecord);
+      // convert srs record to marc and append generated fields
+      marcRecords.add(convert(srsRecord.encode(), mappedFields));
+    }
+    return marcRecords;
+  }
+
+  private List<VariableField> getMappedFields(MappingProfile mappingProfile, String jobExecutionId,
+      OkapiConnectionParams connectionParams, JsonObject srsRecord) {
+    List<VariableField> mappedFields = Collections.emptyList();
+    JsonObject externalIdsHolder = srsRecord.getJsonObject("externalIdsHolder");
+    if (externalIdsHolder != null) {
+      String instanceId = externalIdsHolder.getString("instanceId");
+      if (isNotBlank(instanceId)) {
+        JsonObject holdingsAndItems = fetchHoldingsAndItems(mappingProfile, instanceId, connectionParams);
+        LOGGER.debug("Processing mapping for appending to SRS records for instanceID: {}", instanceId);
+        mappedFields = mappingService.mapFields(holdingsAndItems, mappingProfile, jobExecutionId, connectionParams);
+      }
+    }
+
+    return mappedFields;
+  }
+
+  private JsonObject fetchHoldingsAndItems(MappingProfile mappingProfile, String instanceId,
```

Comment:
It seems mostly a similar method exists in ExportManagerImpl, maybe it's better to extract it to separate class together with the method isTransformationRequired?

Reply:
yup moved to a separate class common to both SRS and inventory.